test/src/integration/integration_test.go

```diff
@@ -81,13 +81,18 @@ func TestEndToEndK8s(t *testing.T) {
 
 		scalarModules := []string{"kubernetes", "cassandra", "network"}
 
+
+
 		for _, m := range scalarModules {
 			terraformOptions := &terraform.Options{
 				TerraformDir: *terraformDir + *cloudProvider + "/" + m,
 				Vars:         map[string]interface{}{},
 				NoColor:      true,
 			}
-
+			if m == "kubernetes" {
+				terraform.RunTerraformCommand(t, terraformOptions, "init")
```

Comment:
I don't think it's necessary here since I think you've already done the `terraform init` process.
https://github.com/scalar-labs/scalar-terratest/blob/ecbf1fc3056574cd8763e5e86dd8d5f0599823cf/test/src/integration/integration_ test.go#L114

Reply:
I have done some test and actually it is.
Before my commit `adding terraform init` the test where failing because it could not find state file.

Reply:
Thank you very much for your understanding. :bow:

Reply:
Do we need `init` here? or we don't need it.
If we need it, it is weird since it's already initialized as Yuji pointed out, so can you clarify? (why do we need it?)

The goal of this is to understand what is going on to fix the root cause, not to make it somehow work.

Reply:
Can you reply back to the comment?
I know you mentioned something in the daily meeting but it should be explained properly here.

So, my understanding `InitAndApply` and `RunTerraformCommand` use different contexts (like two different instances?) so it requires the init again?

Reply:
Actually after multiple test `init` is not needed.
I deleted the empty `Vars` optional object.
Goterratest is checking multiple option and by setting `Vars` to null it was reseting the instance.